pool/pool.go

```diff
@@ -1770,6 +1770,12 @@ func (p *Pool) NetworkInfo(ctx context.Context) (*netmap.NetworkInfo, error) {
 	return cp.client.networkInfo(ctx, prmNetworkInfo{})
 }
 
+// GetCachedEpoch return current epoch from pool cache.
+// It is updated on each request.
+func (p *Pool) GetCachedEpoch() uint64 {
```

Comment:
It seems to me that the pool becomes more complicated with this, I would first try alternative approaches. For example, use an external epoch state. Can you give an example of usage?

Reply:
We wanted to get `CreationEpoch` for the newly created object without an extra `HEAD` request to NeoFS. Currently `s3-gw` uses `pool.NetworkInfo` to do this.